examples/nlp/multimodal_entailment.py

```diff
@@ -0,0 +1,577 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import os
+
+import tensorflow as tf
+import tensorflow_hub as hub
+import tensorflow_text as text
+from tensorflow import keras
+
+"""
+## Define a label map
+"""
+
+label_map = {"Contradictory": 0, "Implies": 1, "NoEntailment": 2}
+
+"""
+## Read the dataset and apply basic preprocessing
+"""
+
+df = pd.read_csv(
+    "https://github.com/sayakpaul/Multimodal-Entailment-Baseline/raw/main/csvs/tweets.csv"
+)
+df.sample(10)
+
+"""
+The columns we are interested in are the following:
+
+* `text_1`
+* `image_1`
+* `text_2`
+* `image_2`
+* `label`
+
+The entailment task is formulated as the following:
+
+***Given the pairs of (`text_1`, `image_1`) and (`text_2`, `image_2`) do they entail (or
+not entail or contradict) each other?*** 
+
+We have the images already downloaded. `image_1` is downloaded as `id1` as its filename
+and `image2` is downloaded as `id2` as its filename. In the next step, we will add two
+more columns to `df` - filepaths of `image_1`s and `image_2`s. 
+"""
+
+images_one_paths = []
+images_two_paths = []
+
+for idx in range(len(df)):
+    current_row = df.iloc[idx]
+    id_1 = current_row["id_1"]
+    id_2 = current_row["id_2"]
+    extentsion_one = current_row["image_1"].split(".")[-1]
+    extentsion_two = current_row["image_2"].split(".")[-1]
+
+    image_one_path = os.path.join("tweet_images", str(id_1) + f".{extentsion_one}")
+    image_two_path = os.path.join("tweet_images", str(id_2) + f".{extentsion_two}")
+
+    images_one_paths.append(image_one_path)
+    images_two_paths.append(image_two_path)
+
+df["image_1_path"] = images_one_paths
+df["image_2_path"] = images_two_paths
+
+# Create another column containing the integer ids of
+# the string labels.
+df["label_idx"] = df["label"].apply(lambda x: label_map[x])
+
+"""
+## Dataset visualization
+"""
+
+
+def visualize(idx):
+    current_row = df.iloc[idx]
+    image_1 = plt.imread(current_row["image_1_path"])
+    image_2 = plt.imread(current_row["image_2_path"])
+    text_1 = current_row["text_1"]
+    text_2 = current_row["text_2"]
+    label = current_row["label"]
+
+    plt.subplot(1, 2, 1)
+    plt.imshow(image_1)
+    plt.axis("off")
+    plt.title("Image One")
+    plt.subplot(1, 2, 2)
+    plt.imshow(image_1)
+    plt.axis("off")
+    plt.title("Image Two")
+    plt.show()
+
+    print(f"Text one: {text_1}")
+    print(f"Text two: {text_2}")
+    print(f"Label: {label}")
+
+
+random_idx = np.random.choice(len(df))
+visualize(random_idx)
+
+random_idx = np.random.choice(len(df))
+visualize(random_idx)
+
+"""
+## Train/test split
+
+The dataset suffers from
+[class imbalance problem](https://developers.google.com/machine-learning/glossary#class-imbalanced-dataset).
+We can confirm that in the following cell.
+"""
+
+df["label"].value_counts()
+
+"""
+To account for that we will go for a stratified split. 
+"""
+
+# 10% for test
+train_df, test_df = train_test_split(
```

Comment:
Don't use sklearn for this, just do it by hand, it will be more explicit

Reply:
I wanted to but since we are doing a stratified split I went with `sklearn`.